### PR TITLE
[TE] Add graceful error handling for pinot data source

### PR DESCRIPTION
**Pull request by cyenjung**
Previously, the method that fetches filter values drops the entire results if it encounters any errors when fetching the filter values for any one of the dimensions of the dataset.

This fix allows the method to return partial results and hence the frontend could still show time series without being blocked.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotDataSourceDimensionFilters.java [hunk 1]
```diff
@@ -79,17 +83,31 @@ public ThirdEyeResponse call() throws Exception {
...
-        Map<String, String> row = thirdEyeResponse.getRow(metricFunction, i);
-        String dimensionValue = row.get(dimension);
-        values.add(dimensionValue);
+      ThirdEyeResponse thirdEyeResponse = null;
+      try {
+        thirdEyeResponse = entry.getValue().get(TIME_OUT_SIZE, TIME_OUT_UNIT);
+      } catch (ExecutionException e) {
+        LOG.error("Execution error when getting filter for Dataset '{}' in Dimension '{}'.", dataset, dimension, e);
+      } catch (InterruptedException e) {
+        LOG.warn("Execution is interrupted when getting filter for Dataset '{}' in Dimension '{}'.", dataset, dimension, e);
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotDataSourceDimensionFilters.java]: interrupted should break from the loop

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java [hunk 2]
```diff
@@ -246,7 +246,12 @@ public ThirdEyeResultSetGroup executePQL(PinotQuery pinotQuery) throws Execution
     Preconditions
         .checkNotNull(this.pinotResponseCache, "{} doesn't connect to Pinot or cache is not initialized.", getName());
 
-    return this.pinotResponseCache.get(pinotQuery);
+    try {
+      return this.pinotResponseCache.get(pinotQuery);
+    } catch (ExecutionException e) {
+      LOG.error("Failed to execute PQL: {}", pinotQuery.getPql());
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java]: shouldn't this be caught and logged at the highest level?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java [hunk 3]
```diff
@@ -261,9 +266,13 @@ public ThirdEyeResultSetGroup refreshPQL(PinotQuery pinotQuery) throws Execution
...
         .checkNotNull(this.pinotResponseCache, "{} doesn't connect to Pinot or cache is not initialized.", getName());
 
-    pinotResponseCache.refresh(pinotQuery);
-
-    return pinotResponseCache.get(pinotQuery);
+    try {
+      pinotResponseCache.refresh(pinotQuery);
+      return pinotResponseCache.get(pinotQuery);
+    } catch (ExecutionException e) {
+      LOG.error("Failed to refresh PQL: {}", pinotQuery.getPql());
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java]: same

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotDataSourceDimensionFilters.java [hunk 4]
```diff
@@ -28,7 +31,9 @@
  */
 public class PinotDataSourceDimensionFilters {
   private static final Logger LOG = LoggerFactory.getLogger(PinotDataSourceDimensionFilters.class);
-  private static final ExecutorService executorService = Executors.newFixedThreadPool(10);
+  private static final ExecutorService executorService = Executors.newFixedThreadPool(40);
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotDataSourceDimensionFilters.java]: maybe use a cached thread pool at this point. 40 idle threads is a lot.

Comment (cyenjung) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotDataSourceDimensionFilters.java]: Nice suggestion! Changed!

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotDataSourceDimensionFilters.java (hunk 1 above)
Comment (cyenjung) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotDataSourceDimensionFilters.java]: Fixed!

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java (hunk 2 above)
Comment (cyenjung) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java]: It should; however, this is the only place that has the raw PQL, which could be used on Pinot Data Explore for debugging purpose. After logging the PQL, this method rethrows the exception in order to allow higher level methods to handle the exception.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java (hunk 3 above)
Comment (cyenjung) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java]: Same reason as the previous one.

**Pull request merged**